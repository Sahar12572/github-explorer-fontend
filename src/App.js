// App.js
import React from "react";
import { BrowserRouter as Router, Route, Routes } from "react-router-dom";
import SearchPage from "./components/SearchPage";
import UserDetails from "./components/UserDetails";

function App() {
  return (
    <Router>
      <div style={{ padding: "2rem", fontFamily: "Arial, sans-serif" }}>
        <h1>GitHub Explorer ðŸ’»âœ¨</h1>
        <Routes>
          <Route path="/" element={<SearchPage />} />
          <Route path="/user/:username" element={<UserDetails />} />
        </Routes>
      </div>
    </Router>
  );
}

export default App;
