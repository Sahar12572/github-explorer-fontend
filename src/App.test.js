import "@testing-library/jest-dom";
import fetchMock from "jest-fetch-mock";
import React from "react";
import { render, screen, fireEvent, waitFor } from "@testing-library/react";
import App from "./App";

fetchMock.enableMocks();

beforeEach(() => {
  fetchMock.resetMocks();
});

// Snapshot test — checks UI doesn’t change unexpectedly
test("renders GitHub Explorer header", () => {
  const { asFragment } = render(<App />);
  expect(asFragment()).toMatchSnapshot();
});

// Unit test for SearchForm and fetch call simulation
test("allows user to search and shows loading", async () => {
  global.fetch = jest.fn(() =>
    Promise.resolve({
      ok: true,
      json: () =>
        Promise.resolve({
          login: "testuser",
          avatar_url: "https://example.com/avatar.png",
          name: "Test User",
          bio: "Hello there",
          html_url: "https://github.com/testuser",
        }),
    })
  );

  render(<App />);

  // Type username
  fireEvent.change(screen.getByPlaceholderText(/enter github username/i), {
    target: { value: "testuser" },
  });

  // Click search button
  fireEvent.click(screen.getByText(/search/i));

  // Check loading text appears
  expect(screen.getByText(/loading user data/i)).toBeInTheDocument();

  // Wait for user data to show
  await waitFor(() => screen.getByText(/test user/i));

  expect(screen.getByText(/hello there/i)).toBeInTheDocument();
  expect(screen.getByRole("img")).toHaveAttribute(
    "src",
    "https://example.com/avatar.png"
  );

  // Cleanup mock
  global.fetch.mockRestore();
});
