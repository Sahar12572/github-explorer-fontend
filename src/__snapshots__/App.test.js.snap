// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders GitHub Explorer header 1`] = `
<DocumentFragment>
  <div
    style="padding: 2rem; font-family: Arial, sans-serif;"
  >
    <h1>
      GitHub Explorer ðŸ’»âœ¨
    </h1>
    <p
      style="color: green;"
    >
      Message from backend: 
    </p>
    <form>
      <input
        placeholder="Enter GitHub username"
        type="text"
        value=""
      />
      <button
        type="submit"
      >
        Search
      </button>
    </form>
  </div>
</DocumentFragment>
`;
